tests/utilities/test_fetching.py

```diff
@@ -174,15 +175,22 @@ def test_dataloader(self):
 def test_trainer_num_prefetch_batches(tmpdir):
 
     model = RecommenderModel()
-    trainer_kwargs = dict(default_root_dir=tmpdir, max_epochs=1, gpus=1, limit_train_batches=3, limit_val_batches=0)
-
-    t0 = time()
-    trainer = Trainer(**trainer_kwargs)
-    trainer.data_connector.data_fetcher = InterBatchParallelDataFetcher()
-    trainer.fit(model)
-    t1 = time()
-    global_step = trainer.global_step
-    assert isinstance(trainer.data_connector.data_fetcher, InterBatchParallelDataFetcher)
+    trainer_kwargs = dict(
+        default_root_dir=tmpdir,
+        max_epochs=1,
+        gpus=1,
+        limit_train_batches=4,
+        limit_val_batches=0,
+        num_sanity_val_steps=0,
+    )
+
+    with mock.patch.dict(os.environ, {"PL_INTER_BATCH_PARALLELISM": "1"}):
+        t0 = time()
+        trainer = Trainer(**trainer_kwargs)
```

Comment:
should the trainer creation be timed too?

Reply:
I don't think this is an issue as it is counted on both side.